pysts/embedding.py

```diff
@@ -101,40 +94,61 @@ def __init__(self, N=300, w2vpath='GoogleNews-vectors-negative%d.bin.gz'):
 
 
 class SkipThought(Embedder):
-    def __init__(self, datadir, uni_bi="combined"):
-        """ Embed Skip_Thought vectors, using precomputed model in npy format.
-
-        Args:
-            uni_bi: possible values are "uni", "bi" or "combined" determining what kind of embedding should be used.
-
-
-        todo: is argument ndim working properly?
-        """
+    """Embedding of sentences, using precomputed skip-thought model [1506.06726].
+    To set up:
+    * Get skipthoughts.py file from https://github.com/ryankiros/skip-thoughts
+    * Execute the "Getting started" wgets in its README
+    * set up config['skipthoughts_datadir'] with path to dir where these files
+        were downloaded
+    
+    Skip-thoughts use embeddings build from the Children Book dataset.
+
+    Config:
+    * config['skipthoughts_uni_bi'] = 'uni' or 'bi' or 'combined'; Two different 
+        skipthought versions, or their combination (see original paper for details)"""
+
+    def __init__(self, c=None):
+        """Load precomputed model."""
+        if not c:
+            c = {}
+        self.c = c
 
         import skipthoughts
         self.encode = skipthoughts.encode
 
-        if datadir is None:
-            datadir = os.path.realpath('__file__')
-        self.datadir = self.datadir
+        if self.c.get("skipthoughts_datadir"):
+            datadir = self.c["skipthoughts_datadir"]
+        else:
+            raise KeyError("config['skipthoughts_datadir'] is not set")
 
         # table for memoizing embeddings
         self.cache_table = {}
 
-        self.uni_bi = uni_bi
-        if uni_bi in ("uni", "bi"):
+        self.uni_bi = self.c["skipthoughts_uni_bi"]
+        if self.uni_bi in ("uni", "bi"):
             self.N = 2400
-        elif uni_bi == "combined":
+        elif self.uni_bi == "combined":
             self.N = 4800
         else:
-            raise ValueError("uni_bi has invalid value. Valid values: 'uni', 'bi', 'combined'")
+            raise KeyError("config['skipthoughts_uni_bi'] has invalid value. Possible values: 'uni', 'bi', 'combined'")
```

Comment:
I think it's ValueError, rather. ;)  (Okay, this *really* is nitpicking, sorry!)